src/core.c/CompUnit/RepositoryRegistry.pm6

```diff
@@ -255,19 +258,27 @@ class CompUnit::RepositoryRegistry {
 
     method repository-for-name(str $name) {
         $*REPO; # initialize if not yet done
-        nqp::ifnull(nqp::atkey($custom-lib,$name),Nil)
+        $lock.protect: {
+            nqp::ifnull(nqp::atkey($custom-lib,$name),Nil)
+        }
     }
 
+    # this feels like dead code  XXX
```

Comment:
This comment is mistaken. For example -- https://github.com/ugexe/CompUnit--Repository--Tar/blob/47021114ba3dbfd9c44c307d300853d7fc773fab/lib/CompUnit/Repository/Tar.rakumod#L16

Reply:
Addressed in https://github.com/rakudo/rakudo/commit/f52e4b995c